packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SingleImageViewer/SingleImageCanvas.js

```diff
@@ -0,0 +1,141 @@
+import { arrayOf, bool, func, number, shape, string } from 'prop-types'
+import { useRef } from 'react'
+
+import SVGContext from '@plugins/drawingTools/shared/SVGContext'
+
+import InteractionLayer from '../InteractionLayer'
+
+import SVGImage from '../SVGComponents/SVGImage'
+
+function calculateAdjustedViewBox({ naturalWidth, naturalHeight, transformMatrix }) {
+  const { scaleX, translateX, translateY } = transformMatrix
+  const scale = scaleX
+  
+  // Calculate new dimensions
+  const newWidth = naturalWidth / scale
+  const newHeight = naturalHeight / scale
+  
+  // Calculate new origin to keep center point
+  const x = (-translateX / scale) 
+  const y = (-translateY / scale)
+  
+  return `${x} ${y} ${newWidth} ${newHeight}`
+} 
+
+function calculateScale({ canvas, naturalWidth }) {
+  const { width: clientWidth } = canvas?.getBoundingClientRect() || {}
+  const calculatedScale = clientWidth / naturalWidth
+  return !Number.isNaN(calculatedScale) ? calculatedScale : 1
+}
+
+const DEFAULT_HANDLER = () => true
+
+function SingleImageCanvas({
+  children,
+  enableInteractionLayer = false,
+  frame = 0,
+  imgRef,
+  invert = false,
+  move = false,
+  naturalHeight,
+  naturalWidth,
+  onDrag = DEFAULT_HANDLER,
+  onKeyDown = DEFAULT_HANDLER,
+  rotation = 0,
+  src,
+  subject,
+  transformMatrix // per VisXZoom
+}) {
+  const canvasLayer = useRef()
+  const canvas = canvasLayer.current
+  
+  const viewBox = `0 0 ${naturalWidth} ${naturalHeight}`
+  
+  const adjustedViewBox = calculateAdjustedViewBox({ naturalWidth, naturalHeight, transformMatrix })
+
+  const rotationTransform = rotation ? `rotate(${rotation} ${naturalWidth / 2} ${naturalHeight / 2})` : ''
+
+  const scale = calculateScale({ canvas, naturalWidth })
+
+  return (
+    <svg
+      onKeyDown={onKeyDown}
+      viewBox={viewBox}
+    >
+      <SVGContext.Provider
+        value={{
+          canvas,
+          rotate: rotation,
+          viewBox: adjustedViewBox,
+          width: naturalWidth,
+          height: naturalHeight,
+        }}
+      >
+        <g
+          data-testid='single-image-canvas-rotation-group'
+          transform={rotationTransform}
+        >
+          <svg
+            ref={canvasLayer}
```

Comment:
This ref is causing the broken dragging behaviour in Firefox. I've removed it in #6491.